# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/TabChild.cpp
# Commit: 52e8c16849b3
# Full Hash: 52e8c16849b3a871a3ea6434c884d32af568337d
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-04 04:33:36
# Regressor Bug: 1510569
# File Overlap Count: 1
# Description:
#   Bug 1510569 - Port onRefreshAttempted from WebProgressChild.jsm to C++ r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D25650
# ==============================================================================

diff -r e1fbefe38944 -r 52e8c16849b3 dom/ipc/TabChild.cpp
--- a/dom/ipc/TabChild.cpp	Wed Apr 03 17:32:41 2019 +0000
+++ b/dom/ipc/TabChild.cpp	Wed Apr 03 17:32:56 2019 +0000
@@ -540,9 +540,9 @@
   nsCOMPtr<nsIDocShell> docShell = do_GetInterface(WebNavigation());
   MOZ_ASSERT(docShell);
 
-  const uint32_t notifyMask = nsIWebProgress::NOTIFY_PROGRESS |
-                              nsIWebProgress::NOTIFY_STATUS |
-                              nsIWebProgress::NOTIFY_CONTENT_BLOCKING;
+  const uint32_t notifyMask =
+      nsIWebProgress::NOTIFY_PROGRESS | nsIWebProgress::NOTIFY_STATUS |
+      nsIWebProgress::NOTIFY_REFRESH | nsIWebProgress::NOTIFY_CONTENT_BLOCKING;
 
   mStatusFilter = new nsBrowserStatusFilter();
 
@@ -3373,7 +3373,10 @@
 NS_IMETHODIMP TabChild::OnRefreshAttempted(nsIWebProgress* aWebProgress,
                                            nsIURI* aRefreshURI, int32_t aMillis,
                                            bool aSameURI, bool* aOut) {
-  return NS_ERROR_NOT_IMPLEMENTED;
+  NS_ENSURE_ARG_POINTER(aOut);
+  *aOut = true;
+
+  return NS_OK;
 }
 
 nsresult TabChild::PrepareProgressListenerData(